src/kdbindings/signal.h

```diff
@@ -294,17 +175,30 @@ class Signal
                 if (index) {
                     const auto con = m_connections.get(*index);
 
-                    if (!con->blocked)
-                        con->slot(p...);
+                    if (!con->blocked) {
+                        if (con->slotDeferred) {
+                            // Check if shared_from_this is safe to call
+                            if (auto sharedThis = shared_from_this(); sharedThis) {
+                                ConnectionHandle handle(sharedThis, *index);
+                                con->slotDeferred(handle, p...);
+                            }
+                        } else if (con->slot) {
+                            con->slot(p...);
+                        }
+                    }
                 }
             }
         }
 
     private:
+        friend class Signal;
         struct Connection {
             std::function<void(Args...)> slot;
+            std::function<void(const ConnectionHandle &, Args...)> slotDeferred;
+            std::weak_ptr<ConnectionEvaluator> m_connectionEvaluator; // Use shared_ptr for stronger ownership.
```

Comment:
What is the `shared_ptr` comment about?
Is it meant as a TODO?

I can't see any reason why we'd want to use shared_ptr. If the connection evaluator is already deleted, we don't need to disconnect from it anymore, and we don't want to queue new invocations into it either, they can't be evaluated anymore anyhow.